src/NuGet/Microsoft.Net.Compilers.Toolset/Microsoft.Net.Compilers.Toolset.Package.csproj

```diff
@@ -23,7 +23,7 @@
     <!-- Remove NU5128 once https://github.com/NuGet/Home/issues/8583 is fixed -->
     <NoWarn>$(NoWarn);NU5100;NU5128</NoWarn>
 
-    <_DependsOn Condition="'$(TargetFramework)' == 'net472'">InitializeDesktopCompilerArtifacts</_DependsOn>
+    <_DependsOn Condition="'$(TargetFramework)' == 'net472' and '$(DotNetBuildFromSource)' != 'true'">InitializeDesktopCompilerArtifacts</_DependsOn>
```

Comment:
This change renders the package incomplete. The build logic in there assumes that the directory and files exist. We have never tested it in a mode where they do not exist. 

Rather than removing this we should dig into why the package isn't working. In your PR you mentioned that Roslyn.Extensions isn't restoring / building here. That shouldn't be necessary though. The `System.Numerics.Vectors.dll` should be coming from our `csi` builds which should be building in source build (to my understanding at least). Not sure how the `Roslyn.Extensions` project gets involved for that DLL